dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/utils/DimensionDataCloudControlResponseUtils.java

```diff
@@ -33,20 +37,41 @@ public static String generateFirewallRuleName(String serverId) {
       return String.format("fw.%s", convertServerId(serverId));
    }
 
-   public static void waitForNetworkDomainStatus(NetworkApi api, String networkDomainId, State state,
-         long timeoutMillis, String message) {
-      boolean isNetworkDomainInState = retry(new NetworkDomainStatus(api, state), timeoutMillis).apply(networkDomainId);
+   public static void waitForNetworkDomainState(NetworkApi api, String networkDomainId, State state, long timeoutMillis,
+         String message) {
+      boolean isNetworkDomainInState = retry(new NetworkDomainState(api, state), timeoutMillis).apply(networkDomainId);
       if (!isNetworkDomainInState) {
          throw new IllegalStateException(message);
       }
    }
 
-   public static void waitForVlanStatus(NetworkApi api, String vlanId, State state, long timeoutMillis,
+   public static void waitForVlanState(NetworkApi api, String vlanId, State state, long timeoutMillis, String message) {
+      boolean isVlanInState = retry(new VlanState(api, state), timeoutMillis).apply(vlanId);
+      if (!isVlanInState) {
+         throw new IllegalStateException(message);
+      }
+   }
+
+   public static void waitForServerState(ServerApi api, String serverId, State state, long timeoutMillis,
          String message) {
-      boolean isVlanInstate = retry(new VlanStatus(api, state), timeoutMillis).apply(vlanId);
-      if (!isVlanInstate) {
+      boolean isServerInState = retry(new ServerState(api, state), timeoutMillis).apply(serverId);
+      if (!isServerInState) {
+         throw new IllegalStateException(message);
+      }
+   }
+
+   public static void waitForServerStatus(ServerApi api, String serverId, boolean started, boolean deployed,
+         long timeoutMillis, String message) {
+      boolean serverHasStatus = retry(new ServerStatus(api, started, deployed), timeoutMillis).apply(serverId);
+      if (!serverHasStatus) {
          throw new IllegalStateException(message);
       }
    }
 
+   public static void waitForVmToolsRunning(ServerApi api, String serverId, long timeoutMillis, String message) {
+      boolean vmwareToolsRunning = retry(new VMToolsRunningStatus(api), timeoutMillis).apply(serverId);
+      if (!vmwareToolsRunning) {
+         throw new IllegalStateException(message);
+      }
+   }
```

Comment:
Instead of having a static class with these accessors, consider configuring all these predicates as injectable predicates. this makes it easier to get them injected in the jclouds classes (such as the compute service when implemented), and also allows users to configure the timeout values, etc, via properties.
You can have a look at [the digitalocean2 provider](https://github.com/jclouds/jclouds/blob/master/providers/digitalocean2/src/main/java/org/jclouds/digitalocean2/compute/config/DigitalOcean2ComputeServiceContextModule.java#L100-L216) for an example.

Reply:
I understand what you are looking for. I will not make the change yet since we have not added our DimensionDataCloudControlServiceContextModule instance yet. When this is added (maybe next week) I will add this change too.